plugins/modules/source_control/gitlab/gitlab_group_members.py

```diff
@@ -165,8 +183,9 @@ def main():
     argument_spec.update(dict(
         api_token=dict(type='str', required=True, no_log=True),
         gitlab_group=dict(type='str', required=True),
-        gitlab_user=dict(type='str', required=True),
-        state=dict(type='str', default='present', choices=['present', 'absent']),
+        gitlab_user=dict(type='str', required=False),
+        gitlab_users=dict(type='list', elements='str', required=False),
+        state=dict(type='str', default='present', choices=['present', 'absent', 'present-exact']),
```

Comment:
I would avoid such a state name. Instead use `state=present` with another option which specifies that what should happen to users that are not on the list, like a boolean option `purge_users` or something like that.

Also I'm not sure whether this is a good interface, since generally you don't want that all users have exactly the same access level. If you want to have users A and B as maintainers and C, D and E as developers, and nobody else, you cannot do that with the interface you are proposing, since you can either make sure there's nobody else on the list, or you can have different access rights for A, B vs. C, D, E.

Reply:
I think you can. You'd have to call the module for each access level with a user list. So in your example you'd call
gitlab_group_members:
  gilab_users:
    - 'A'
    - 'B'
  access_level: 'developer'
  purge_users: true

gitlab_group_members:
  gilab_users:
    - 'C'
    - 'D'
    - 'E'
  access_level: 'maintainer' 
  purge_users: true

Reply:
Ok, but that only works if `purge_users` only applies to users with this access level.

I'm right now wondering a bit whether the interface shouldn't be a list, but a dictionary mapping username to access level. That way, you can provide multiple users for one access level and manage all memberships in one go.

Reply:
@Lunik @Shaps @dj-wasabi @marwatk @metanovii @scodeman @waheedi @zanssa if you have ideas how a good interface could look which manages multiple (or all) users at the same time, please feel free to chime in :)